minetest/minetest
# Newirrlicht

With the CMake changes to IrrlichtMt, it's now possible to use a target for IrrlichtMt, besides greatly improving the ease of setting up IrrlichtMt for users building the client. It also removes the need for Minetest's CMake to include PNG and JPEG, cleaning it up a tiny bit.
        The PR works by finding the IrrlichtMt package and linking to the target it provides. If the package isn't found and it isn't building the client, it will still fall back to using just the headers with the user provided path.

## To do

This PR is a Work in Progress

- [ ] If the target isn't found, currently it will still fail to build.

## How to test

- Build and install IrrlichtMt, for example:
- `cmake -S . - B build -DBUILD_SHARED_LIBS=OFF -DCMAKE_INSTALL_PREFIX=~/.local`
- `cmake --build build -j(nproc)`
- `cmake --install build`
- Build Minetest, for example:
- `cmake -S . -B build -DRUN_IN_PLACE=TRUE -DCMAKE_PREFIX_PATH=~/.local`
- `cmake --build build -j(nproc)`
- Observe that it builds and runs correctly.
